chapters/missing-data.qmd

```diff
@@ -0,0 +1,598 @@
+  mutate(`Percent Missing` = 100 * value / nrow(ames))
+
+#Remove predictors with more than 20% missing data
+hotels_pred_gt_20 <-
+  hotels_missing_summary %>%
+  dplyr::filter(`Percent Missing` > 20) %>%
+  dplyr::select(Predictors) %>%
+  pull()
+
+hotels_predictors_subset <-
+  hotels_predictors %>%
+  dplyr::select(-all_of(hotels_pred_gt_20))
+
+hotels_pct_missing_subset <- 100 * 
+  sum(is.na(hotels_predictors_subset)) /
+  (nrow(hotels_predictors_subset) * ncol(hotels_predictors_subset))
+
+# Compute the percent missing values within each sample
+hotels_samples <-
+  hotels_predictors_subset %>%
+  mutate(RowNum = row_number()) %>%
+  mutate(across(everything(), as.character)) %>%
+  pivot_longer(cols=-RowNum, names_to = "Predictor", values_to = "value") %>%
+  mutate(Missing = ifelse(is.na(value), "Yes", "No")) %>%
+  mutate(Missing = factor(Missing, levels=c("Yes", "No"))) %>%
+  dplyr::select(-value) %>%
+  pivot_wider(id_cols="Predictor", names_from="RowNum", values_from="Missing") %>%
+  dplyr::select(-Predictor) %>%
+  mutate(across(everything(), binary_fcn)) %>%
+  summarize_all(~sum(.)/ncol(hotels_predictors_subset)*100)
+
+max_missing_hotels_samples <- max(hotels_samples)
+```
+
+When collecting data for analysis, it is not uncommon for some predictor values to be unknown. There can be a multitude of reasons that this can happen. For example, when a laboratory test for a respiratory disease is included in a data set, a sample may have no result for a few reasons. First, there might have been a data collection error; someone forgot to include the value. It could also be due to a database merge where a key field was unspecified and a subset of the data was not properly joined. It is also possible that the test itself failed to produce a result. For example, the test might fail for diagnostics that use throat swabs because an interfering substance, such as food coloring from a lozenge, is on the swab. 
+
+As another example, two data sets used throughout this text, houses in Ames and hotel rates, contain missing data. The Ames Housing data has `r round(ames_pct_missing, 1)`% missing values across `r ames_num_missing_predictors` predictors, while the hotel rate data has `r round(hotels_pct_missing, 1)`% missing values across `r hotels_num_missing_predictors` predictors. @fig-ames-missing illustrates the occurrence of missingness for the Ames data. This figure presents the full rectangular data matrix where each property (sorted alphabetically by neighborhood) is on the x-axis, and each predictor is on the y-axis. Red indicates that the value was missing in the data, while gray indicates that the value was not missing. 
+
+```{r}
+#| label: fig-ames-missing
+#| echo: false
+#| out-width: 100%
+#| fig-width: 10.5
+#| fig-height: 10.5
+#| fig-cap: A visualization of the missing data for the Ames Housing data set.
+
+ggplot(ames_missing_long, aes(x = RowNum, y = Predictor, fill = Missing)) + 
```

Comment:
This will be more clear if we only show the predictors with missing data.

Reply:
I'd also like to add an ["upset plot"](https://naniar.njtierney.com/reference/gg_miss_upset.html) to this figure (or add another figure).

Reply:
Good points.  Subsetting to just the predictors with missing data and adding an upset plot will provide better insights to the missing information.  I can do this.

Reply:
I had worked on this last week and checked it in. Go ahead and edit/update as needed.